### ISSUE TITLE
OData V2 Proxy doesnt set severity in errordetails

### ISSUE THREAD

**Issue by gregorwolf** [sapmentors/cap-community#70]
As Fiori Elements isn't fully supported with OData V4 I'm using the @sap/cds-odata-v2-adapter-proxy to provide our CAP service as OData V2. But there is an issue how the proxy translates error messages. This results in the issue that the error message isn't shown in the Fiori Elements UI.

With an OData V4 Fiori Elements UI the error message is shown like this:

![Bildschirmfoto vom 2020-05-07 22-50-09](https://user-images.githubusercontent.com/2639535/81345566-fb235380-90b8-11ea-9fc5-94c60e45ffa7.png)

The service response data looks like that:

```
--batch_id-1588885046769-444
content-type: application/http

HTTP/1.1 403 Forbidden
odata-version: 4.0
content-type: application/json;odata.metadata=minimal;IEEE754Compatible=true

{
  "error": {
    "code": "403",
    "message": "user error@example.com isn't assigned to any role"
  }
}
--batch_id-1588885046769-444--
```

When the same service is called using OData V2 then the payload looks like that:

```
--batch_6266-3353-cf2d
content-type: application/http

HTTP/1.1 403 Forbidden
content-type: application/json
dataserviceversion: 2.0
content-length: 239

{
  "error": {
    "code": "403",
    "message": {
      "lang": "en",
      "value": "user error@example.com isn't assigned to any role"
    },
    "innererror": {
      "errordetails": [
        {
          "code": "403",
          "message": {
            "lang": "en",
            "value": "user error@example.com isn't assigned to any role"
          }
        }
      ]
    }
  }
}
--batch_6266-3353-cf2d--
```

But the user only sees:

![Bildschirmfoto vom 2020-05-07 23-08-41](https://user-images.githubusercontent.com/2639535/81345679-2efe7900-90b9-11ea-8bac-5ce0a1cbc7ce.png)

When I create an error message in OData V2 with SAP Gateway (ABAP) the payload is this:

```
--2CE3BCE5BF4CB972AA7146308FF60B6A0
Content-Type: application/http
Content-Length: 1022
content-transfer-encoding: binary

HTTP/1.1 400 Bad Request
Content-Type: application/json;charset=utf-8
Content-Length: 902
dataserviceversion: 1.0

{
  "error": {
    "code": "SY/002",
    "message": {
      "lang": "de",
      "value": "user reader isn't assigned to any role."
    },
    "innererror": {
      "application": {
        "component_id": "",
        "service_namespace": "/SAP/",
        "service_id": "ZEPM_BP_SRV",
        "service_version": "0001"
      },
      "transactionid": "8280D1D62E110020E005EB30A6FB1945",
      "timestamp": "20200507210440.7358260",
      "Error_Resolution": {
        "SAP_Transaction": "For backend administrators: run transaction /IWFND/ERROR_LOG on SAP Gateway hub system and search for entries with the timestamp above for more details",
        "SAP_Note": "See SAP Note 1797736 for error analysis (https://service.sap.com/sap/support/notes/1797736)"
      },
      "errordetails": [
        {
          "code": "/IWBEP/CX_MGW_BUSI_EXCEPTION",
          "message": "user reader isn't assigned to any role",
          "propertyref": "",
          "severity": "error",
          "target": ""
        }
      ]
    }
  }
}
--2CE3BCE5BF4CB972AA7146308FF60B6A0--
```

And the error is shown as:

![Bildschirmfoto vom 2020-05-07 23-21-40](https://user-images.githubusercontent.com/2639535/81345907-97e5f100-90b9-11ea-8093-69862a925ac2.png)

We've debugged through the SAPUI5 code and the reason is that the attribute "severity": "error" is missing in the OData V2 response generated by the proxy. This checks are done in the _parseBodyJSON function of the [ODataMessageParser.js#L729](https://github.com/SAP/openui5/blob/744e910fd2d74f83f229150150ce141d105661d3/src/sap.ui.core/src/sap/ui/model/odata/ODataMessageParser.js#L729).

You can replicate this issue by using the current version of my [/bookshop-demo](https://github.com/gregorwolf/bookshop-demo) application. Execute the setup as described in the readme. Then start the app also as described. In the Launchpad start the Browse Books App from the V2 Group. Login with user "error" to see the issue.

**Comment by gregorwolf** [comment]
Also filed via the SAP Service Marketplace 307914 / 2020.

**Comment by gregorwolf** [comment]
Thank you Oliver. The issue is fixed in CDS OData v2 Adapter Proxy 1.4.30.
